examples/tinywl-new/output.cpp

```diff
@@ -341,9 +335,37 @@ void Output::layoutNonLayerSurface(SurfaceWrapper *surface, const QSizeF &sizeDi
             if (normalGeo.size().isEmpty())
                 return;
 
-            normalGeo.moveCenter(validGeo.center());
-            normalGeo.moveTop(qMax(normalGeo.top(), validGeo.top()));
-            normalGeo.moveLeft(qMax(normalGeo.left(), validGeo.left()));
+            auto parentSurface = surface->shellSurface()->parentSurface();
+            SurfaceWrapper* parentSurfaceWrapper = nullptr;
+            if (parentSurface) {
```

Comment:
If this surface have a parent, you should use `SurfaceWrapper::addSubSurface` for it in `Helper` class.